dtfs-central-api/src/helpers/calculate-fixed-fee-from-days-remaining.test.ts

```diff
@@ -0,0 +1,29 @@
+import Big from 'big.js';
+import { calculateFixedFeeFromDaysRemaining } from './calculate-fixed-fee-from-days-remaining';
+
+describe('calculateFixedFeeFromDaysRemaining', () => {
+  it('should return the product of the utilisation, interest percentage (divided by 100), bank admin fee percentage (fixed at 0.9) and number of days remaining in the cover period divided by the day count basis and rounds to 2 decimal places', () => {
+    // Arrange
+    const utilisation = 100000;
+    const interestPercentage = 5;
+    const dayCountBasis = 365;
+
+    // Number of days remaining = 432
+    const numberOfDaysRemainingInCoverPeriod = 432;
+
+    // Act
+    const result = calculateFixedFeeFromDaysRemaining({
+      utilisation,
+      numberOfDaysRemainingInCoverPeriod,
+      interestPercentage,
+      dayCountBasis,
+    });
+
+    // Assert
+    // utilisation * (interestPercentage / 100) * BANK_ADMIN_FEE_ADJUSTMENT * numberOfDaysRemainingInCoverPeriod / dayCountBasis = 5,326.02739... = 5,326.03 (ROUNDED)
+    const calculation = (utilisation * (interestPercentage / 100) * 0.9 * numberOfDaysRemainingInCoverPeriod) / dayCountBasis;
```

Comment:
Can we just assert that it's equal to 5326.03 please as we know the answer? In this case I want the test to check it's giving the right answer, not re-write the implementation and then check the implementation in the controller is the same as the implementation written in the test, I just care about the end result

Reply:
I have changed this.  I do not fully agree with this as if we change the mocks, then this test will fail and it will be hard to see why the test is failing for someone new.  Whereas with the old implementation, it will be resilient to mock changes but a change in function will cause it to fail (as the calculation will change)

Reply:
Thanks, can completely see your point of view but for a calculation test like this I don't mind it being a bit brittle if it increases my confidence that the test is doing the exact right calculation and output . We have some other tests closer to what you've got so at least we have a mix to cover everything now :D